src/rt/sections_elf_shared.d

```diff
@@ -461,8 +461,6 @@ extern(C) void _d_dso_registry(CompilerDSOData* data)
         else
         {
             // static DSOs are unloaded in reverse order
-            assert(pdso._tlsSize == _tlsRanges.back.length);
-            _tlsRanges.popBack();
```

Comment:
Before this PR, yes. With this PR it will have no elements at that time (and the `.popBack` will thus result in an `AssertError`, which is why I removed it), because it was already `.clear`ed via `thread_term` (as per PR description).